### PR TITLE
Server test setup cleanup

### PR DESCRIPTION
**Pull request by codyebberson**
1. Fix tests that used `project: randomUUID()` - this was a quick hack that we used to create pseudo-project isolation, but the recent changes to subscription access policies exposed this.
2. Updated the `createTestProject()` helper
    a. Only create `client`, `login`, `accessPolicy`, and `repo` conditionally
3. Added `getLogger()` helper to replace `getRequestContext().logger`
    a. If in a request context, `getLogger()` returns the equivalent of `getRequestContext().logger`
    b. Otherwise, returns `systemLogger`
    c. This opens up a bunch of different code paths to be used outside of a request context
    d. This dramatically reduces the number of cases where we need to use `withTestContext()`
4. Switched `createTestProject()` to use `requestContextStore.run` rather than `requestContextStore.enterWith`
    a. Using `enterWith` was obfuscating a bunch of context bugs in tests
5. Fixed bug in `Project/$clone`
    a. Need to create the new `Project` resource first, so that all of the follow-up requests can reference the `Project`

### CODE REVIEW

File: packages/server/src/fhir/repo.ts [hunk 1]
```diff
@@ -786,10 +786,8 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
       throw new OperationOutcomeError(forbidden);
     }
 
-    await this.withTransaction(async () => {
-      const resource = await this.readResourceImpl<T>(resourceType, id);
-      return this.reindexResourceImpl(resource);
-    });
+    const resource = await this.readResourceImpl<T>(resourceType, id);
```
Comment (mattlong) [inline packages/server/src/fhir/repo.ts:789]: Why was `withTransaction` removed?

**Review by mattlong — COMMENTED** [review]
Nice cleanups! Really just the one question about whether `getLogger` needs to be able to distinguish an authenticated context.

File: packages/server/src/admin/invite.ts [hunk 2]
```diff
@@ -65,7 +65,8 @@ export interface ServerInviteResponse {
 
 export async function inviteUser(request: ServerInviteRequest): Promise<ServerInviteResponse> {
   const systemRepo = getSystemRepo();
-  const ctx = getAuthenticatedContext();
+  const logger = getLogger();
```
[the remark this answers is not in the document]
Comment (mattlong) [reply packages/server/src/admin/invite.ts:68]: The intent was just to get the logger, but perhaps that check was providing some (unintended?) security?

[the remark this answers is not in the document]
Comment (codyebberson) [reply packages/server/src/admin/invite.ts:68]: Good question.... The public API is handled above in `inviteHandler`, which does call `getAuthenticatedContext()` and uses it to validate the user.  

This `inviteUser()` is more of an internal utility method, which we use in tests.

File: packages/server/src/fhir/repo.ts (hunk 1 above)
Comment (codyebberson) [reply packages/server/src/fhir/repo.ts:789]: Oops, good catch.  There was a test that was failing on this, which was fixed separately.  I'll clean this up.

**Review by mattlong — APPROVED** [review]

**Pull request merged**